### PR TITLE
BE-006-2 (Nikolay Chornyi) Add Repositories & Service & Mapper

### PR DESCRIPTION
**Pull request by mblack666**
Add Repositories & Service interfaces & Mapper (Vendor, Location, Discount, Category)

### CODE REVIEW

**Review by Hanna-Kalashnyk — DISMISSED** [review]

File: src/main/java/com/exadel/discount/service/VendorLocationService.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.List;
+import java.util.UUID;
+
+public interface VendorLocationService {
+
+    LocationDTO save(CreateLocationDTO createLocationDTO);
+
+    LocationDTO getById(UUID id);
+
+    List<LocationDTO> getAll(UUID vendorId);
```
Comment (mikevoloshyn) [inline src/main/java/com/exadel/discount/service/VendorLocationService.java]: please rename method. 
if there is parameter please put it on method name like
getLocationsByVendorId or getLocationsForVendorById, etc.

File: src/main/java/com/exadel/discount/repository/VendorLocationRepository.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.UUID;
+
+@Repository
+public interface VendorLocationRepository extends JpaRepository<VendorLocation, UUID> {
+
+    List<VendorLocation> findByVendor(Vendor vendor);
```
Comment (mikevoloshyn) [inline src/main/java/com/exadel/discount/repository/VendorLocationRepository.java]: Are you actually wanna search by full vendor entity or just specific fk?

**Review by mikevoloshyn — DISMISSED** [review]

File: src/main/java/com/exadel/discount/service/VendorLocationService.java (hunk 1 above)
Comment (mblack666) [reply src/main/java/com/exadel/discount/service/VendorLocationService.java]: +

File: src/main/java/com/exadel/discount/repository/VendorLocationRepository.java (hunk 2 above)
Comment (mblack666) [reply src/main/java/com/exadel/discount/repository/VendorLocationRepository.java]: non actually, edited to search by vendorId

**Review by SergeyKoval — APPROVED** [review]

**Review by SerhiiPakulia — APPROVED** [review]

**Review by Hanna-Kalashnyk — APPROVED** [review]

**Pull request merged**